src/crl.c

```diff
@@ -167,6 +200,46 @@ static int CheckCertCRLList(WOLFSSL_CRL* crl, DecodedCert* cert, int *pFoundEntr
             int doNextDate = 1;
 
             WOLFSSL_MSG("Found CRL Entry on list");
+
+            if (crle->verified == 0) {
+                Signer* ca;
+
+                wc_UnLockMutex(&crl->crlLock);
+
+            #if !defined(NO_SKID) && defined(CRL_SKID_READY)
+                if (crle->extAuthKeyIdSet)
+                    ca = GetCA(crl->cm, crle->extAuthKeyId);
+                if (ca == NULL)
+                    ca = GetCAByName(crl->cm, crle->issuerHash);
+            #else /* NO_SKID */
+                ca = GetCA(crl->cm, crle->issuerHash);
+            #endif /* NO_SKID */
+                if (ca == NULL) {
+                    WOLFSSL_MSG("Did NOT find CRL issuer CA");
+                    return ASN_CRL_NO_SIGNER_E;
+                }
+
+                ret = VerifyCRL_Signature(crle->toBeSigned, crle->tbsSz,
+                    crle->signature, crle->signatureSz, crle->signatureOID, ca);
+
+                if (wc_LockMutex(&crl->crlLock) != 0) {
+                    WOLFSSL_MSG("wc_LockMutex failed");
+                    return BAD_MUTEX_E;
+                }
+
+                if (ret == 0)
+                    crle->verified = 1;
```

Comment:
crle may have been freed while the lock was released.  I feel like we need to make a deep copy of crle to do the verification and after getting the long again see if crle still exists.